### PR TITLE
T3 - Test Log Out button works from every page - first commit

### PR DESCRIPTION
**Pull request by dankolojanica1**

### CODE REVIEW

File: endavaTraining/src/main/java/com/endavatraining/pages/HomePage.java [hunk 1]
```diff
@@ -2,11 +2,25 @@
...
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindAll;
+
+import java.util.List;
 
 
 public class HomePage extends BasePage {
 
+	private By logOutButton = By.xpath("//*[@id=\"headContainer\"]/nav/div/ul[2]/li[2]/a");
```
Comment (BoskoNikolic) [inline endavaTraining/src/main/java/com/endavatraining/pages/HomePage.java]: Why not simply use an ID here?:

File: endavaTraining/src/main/java/com/endavatraining/pages/BasePage.java [hunk 2]
```diff
@@ -1,11 +1,14 @@
 package com.endavatraining.pages;
 
+import org.apache.log4j.Logger;
 import org.openqa.selenium.By;
 import org.openqa.selenium.NoSuchElementException;
 import org.openqa.selenium.WebDriver;
 
 public class BasePage {
 
+	public static Logger logger = Logger.getLogger(BasePage.class);
```
Comment (BoskoNikolic) [inline endavaTraining/src/main/java/com/endavatraining/pages/BasePage.java]: logger is not yet committed so it should not be part of this pull request.

File: endavaTraining/src/main/java/com/endavatraining/pages/HomePage.java [hunk 3]
```diff
@@ -17,4 +31,38 @@ public boolean isWelcomeTextPresent() {
 		return isElementPresent(homePageWelcomeMessage);
 	}
 
+/**
+    * This method clicks on every menu button depending on parameter is passed
+    * @author danko lojanica
+    * @param  button
+ */
+	public void logOutButton(String button){
```
Comment (BoskoNikolic) [inline endavaTraining/src/main/java/com/endavatraining/pages/HomePage.java]: Not all logout buttons are part od home page. Check with Marko if he's fine with this approach or you should create clicklogout from every page class.

File: endavaTraining/src/main/java/com/endavatraining/pages/LoginPage.java [hunk 4]
```diff
@@ -15,15 +18,21 @@
 	public LoginPage(WebDriver driver) {
 		super(driver);
 	}
-	
+
+	public void userLogin(String username, String password) {
```
Comment (BoskoNikolic) [inline endavaTraining/src/main/java/com/endavatraining/pages/LoginPage.java]: Have you merged code with the latest master before you created this pull request?

File: endavaTraining/src/test/java/com/endavatraining/TestHomePage.java [hunk 5]
```diff
@@ -43,6 +52,48 @@ public void testIsHomePageAvailable() {
...
 
+	/*
+	 * Test validates that Log Out button works from every page
+	 */
+	@Test
+	public void testIsLogOutWorks() {
+		loginPage = new LoginPage(new ChromeDriver());
+		homePage = loginPage.openAs(username, password);
+        homePage.logOutButton("Home");
+		new WebDriverWait(loginPage.driver, 5)
```
Comment (BoskoNikolic) [inline endavaTraining/src/test/java/com/endavatraining/TestHomePage.java]: When user is logged out these should be a specific message on UI. Point of this test was to catch that message not login button.

File: endavaTraining/src/main/java/com/endavatraining/pages/HomePage.java (hunk 3 above)
Comment (endavamarko) [reply endavaTraining/src/main/java/com/endavatraining/pages/HomePage.java]: We agreed to put logout method at BasePage

File: endavaTraining/src/test/java/com/endavatraining/TestHomePage.java [hunk 6]
```diff
@@ -41,6 +44,50 @@ public void testIsHomePageAvailable() {
...
 
+	/**
+	 * Test validates that log out button works properly from every page
+	 * @author Danko.Lojanica
+	 */
+
+	@Test
+	public void testIsLogOutWorks() {
+		loginPage = new LoginPage(new ChromeDriver());
+		homePage = loginPage.openAs(username, password);
```
Comment (endavamarko) [inline endavaTraining/src/test/java/com/endavatraining/TestHomePage.java]: Don't hardcode Chrome

File: endavaTraining/src/test/java/com/endavatraining/TestHomePage.java [hunk 7]
```diff
@@ -51,47 +58,43 @@ public void testIsHomePageAvailable() {
...
 	public void testIsLogOutWorks() {
-		loginPage = new LoginPage(new ChromeDriver());
+		loginPage.open();
 		homePage = loginPage.openAs(username, password);
-		homePage.logOutButton("Home");
-		new WebDriverWait(loginPage.driver, 5)
-				.until(ExpectedConditions.visibilityOfElementLocated(logOutMessage));
+		homePage.clickOnButton(homeButton);
+		homePage.clickOnButton(logOutButton);
+		Assert.assertEquals(homePage.driver.findElement(logOutMessage).getText(), BasePage.LOG_OUT_MESSAGE, "Log out failed");
```
Comment (endavamarko) [inline endavaTraining/src/test/java/com/endavatraining/TestHomePage.java]: Don't call this via BasePage, you can call it via HomePage as it is child page. We shouldn't mention basepage at tests

File: endavaTraining/src/test/java/com/endavatraining/TestHomePage.java [hunk 8]
```diff
@@ -51,47 +58,43 @@ public void testIsHomePageAvailable() {
...
 
 		loginPage.userLogin(username,password);
-		homePage.logOutButton("Users");
-		new WebDriverWait(loginPage.driver, 5)
-				.until(ExpectedConditions.visibilityOfElementLocated(logOutMessage));
+		homePage.clickOnButton(usersButton);
+		homePage.clickOnButton(logOutButton);
+		Assert.assertEquals(homePage.driver.findElement(logOutMessage).getText(),  BasePage.LOG_OUT_MESSAGE, "Log out failed");
```
Comment (endavamarko) [inline endavaTraining/src/test/java/com/endavatraining/TestHomePage.java]: same comment as belllow
